### PR TITLE
sorbet: Use Tapioca more since `srb rbi` is in maintenance

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

- I was reading about modern Sorbet and found that `srb rbi` is in maintenance mode. The Sorbet project recommend using Tapioca. They even made a [migration guide](https://github.com/Shopify/tapioca/wiki/Migrating-to-Tapioca). It wasn't that hard to follow given we already use Tapioca in some places.
- According to the migration guide, `hidden-definitions` isn't necessary any more. There's also another gem called Spoom for bumping `typed:` sigils that's recommended over `srb --suggest-typed`. For consistency, I've kept the `brew typecheck` flags the same for now, but it's a bit of a misnomer since Spoom actually does make the changes that we'll have to commit and push.
- After all these changes, `brew typecheck` comes back with no new or existing errors, and we're further along with modern Sorbet tooling adoption.

### CODE REVIEW

**Comment by Bo98** [comment]
> * According to the migration guide, `hidden-definitions` isn't necessary any more.

I'm tad sceptical given we've not gone fully typed yet. You haven't actually removed the file so try do that and see if it's still no errors.

**Review by MikeMcQuaid — APPROVED** [review]

**Review by Bo98 — APPROVED** [review]
